Docs/src/instancing.md

```diff
@@ -0,0 +1,373 @@
+
+Instancing {#instancing}
+==========
+
+@tableofcontents
+
+# What is instancing? {#WhatIsInstancing}
+
+Instancing is a rendering technique to draw multiple instances of the
+same mesh using just one render call. There are two kinds of instancing:
+
+-   **Software:** Two larges vertex & index buffers are created and the mesh vertices/indices are duplicated N number of times. 
+    When rendering, invisible instances receive a transform matrix filled with 0s. 
+	This technique can take a lot of VRAM and has limited culling capabilities.
+-   **Hardware:** The hardware supports an extra param which allows Ogre to tell the GPU to repeat the drawing of vertices N number of times;
+    thus taking considerably less VRAM. Because N can be controlled at runtime, individual instances can be culled before sending the data to the GPU.
+
+Hardware techniques are almost always superior to Software techniques, but Software are more compatible, where as Hardware techniques require D3D9 or GL3, and is not supported in GLES2
+
+All instancing techniques require shaders. 
```

Comment:
this paragraph is confusing with ogre1, as instancing can be transparently handled by the RTSS. just remove it.

Reply:
I think that people shoud know that you require shaders in some way even if the RTSS generates them for you.

So I will put that... but good point.